incrementals-publisher/README.adoc

```diff
@@ -0,0 +1,79 @@
+= Incremental Build Publisher
+
+
+This Azure Function is responsible for bouncing incrementally built (see
+link:https://github.com/jenkinsci/jep/tree/master/jep/305[JEP-305]) core and
+plugin artifacts into the incrementals Maven repository (see
+link:https://github.com/jenkins-infra/iep/tree/master/iep-009[IEP-9]).
+
+
+[[env]]
+== Environment Variables
+
+|===
+| Variable | Description
+
+| `GITHUB_TOKEN`
+| A GitHub OAuth/Personal Access Token which can be used for GitHub API queries
+
+| `JENKINS_HOST`
+| A Jenkins instance (defaults to `https://ci.jenkins.io`) to which URLs are
+expected to confirm.
+
+| `INCREMENTAL_URL`
+| A Maven repository which should be treated as the incrementals destination,
+defaults to `https://repo.jenkins-ci.org/incrementals/`
+
+| `ARTIFACTORY_KEY`
+| An Artifactory user's API key
+
+| `PERMISSIONS_URL`
+| A URL pointing to the generated repository-permissions-updater output,
+defaults to
+`https://ci.jenkins.io/job/Infra/job/repository-permissions-updater/job/master/lastSuccessfulBuild/artifact/json/github-index.json`
+
+|===
+
+=== Optional/Development Variables
+
+|===
+| Variable | Description
+
+| `METADATA_URL`
+| URL which will serve up JSON which represents metadata about a Pipeline Run (link:https://gist.github.com/rtyler/6b601864e676d0f0735c1399e291ddf4#file-gistfile1-txt[example])
+
+| `ARCHIVE_URL`
+| URL to a `.zip` file which represents an example release zip generated by the incrementals release tooling.
+|===
+
+== Valid Requests
+
+This Function only supports one kind of request, and is expected to only be
+called by the `buildPlugin()` step or other Jenkins Pipelines which are
+publishing incrementally built artifacts into Artifactory.
```

Comment:
Worth mentioning that the expected value is the `BUILD_URL` environment variable defined by Jenkins.